filter.h

```diff
@@ -0,0 +1,72 @@
+// Filter Module
+
+
+// ****CHECKPOINT GOALS****
+// USER INTERFACE
+// PSEUDO CODE
+//IMPLEMENTATION
+//WELL DOCUMENTATED HEADER FILE
+// DOXYGEN FILE 
+// TEMPLATE FOR C FILE
+
+
+
+/*
+@param filter_t * - a pointer of struct type filter_t
+@param offset -  describes postion of the data going into an array
```

Comment:
No, add more documentation and mention that when Filter_update is called the raw value goes through linear transfer function new_value = raw_value * scale + offset. Also offset needs to be int16_t.